File: app/src/main/java/com/evernote/android/job/sample/ReminderJobCreator.java
package com.evernote.android.job.sample;

import com.evernote.android.job.Job;
import com.evernote.android.job.JobCreator;
import com.evernote.android.job.sample.reminder.ReminderJob;
import com.evernote.android.job.sample.sync.SyncJob;

/**
 * @author rwondratschek
 */
public class ReminderJobCreator implements JobCreator {

    @Override
    public Job create(String tag) {
        switch (tag) {
            case ReminderJob.TAG:
                return new ReminderJob();

            case SyncJob.TAG:
                return new SyncJob();

            default:
                return null;
        }
    }
}

File: app/src/main/java/com/evernote/android/job/sample/ReminderJobCreator.kt
package com.evernote.android.job.sample

import com.evernote.android.job.Job
import com.evernote.android.job.JobCreator
import com.evernote.android.job.sample.reminder.ReminderJob
import com.evernote.android.job.sample.sync.SyncJob

/**
 * @author rwondratschek
 */
class ReminderJobCreator : JobCreator {

    override fun create(tag: String): Job? {
        return when (tag) {
            ReminderJob.TAG -> ReminderJob()
            SyncJob.TAG -> SyncJob()
            else -> null
        }
    }
}
